core/src/franchise/player/player.controller.ts

```diff
@@ -36,122 +38,127 @@ export class PlayerController {
 
     @Get("accept-rankdown/:token")
     async acceptRankdown(@Param("token") token: string): Promise<string> {
-        const payload = this.jwtService.decode(token) as RankdownPayload | null;
-        if (!payload) return "FAILED TO DECODE PAYLOAD";
+        try {
+            const payload = this.jwtService.verify<RankdownPayload>(token);
+            if (!payload) return "FAILED TO DECODE PAYLOAD";
```

Comment:
We may want this to be a nicer message; because the user will see this directly.